presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -196,6 +200,36 @@ public BlockBuilder closeEntry()
         return this;
     }
 
+    /**
+     * This method will check duplicate keys and close entry.
+     * <p>
+     * When duplicate keys are discovered, the block is guaranteed to be in
+     * a consistent state before {@link DuplicateMapKeyException} is thrown.
```

Comment:
I find `consistent` a bit vague here. Maybe add `In other words, one can continue to use this BlockBuilder.`

In addition, I think it makes sense to make `DuplicateMapKeyException` a checked exception. After all, that's the point of invoking `closeEntryStrict`. If you don't care, why not invoke `closeEntry` instead.